provider/provider.go

```diff
@@ -90,8 +90,14 @@ func (p *dockerNativeProvider) Configure(_ context.Context, req *rpc.ConfigureRe
 		return nil, err
 	}
 	host := client.DaemonHost()
-	log.Printf("Setting DOCKER_HOST to %s", host)
-	os.Setenv("DOCKER_HOST", host)
+
+	// In the case of a remote docker client that we connect to with SSH, we use a connection helper as the daemon host.
+	// Per the docker/cli/cli/connhelper package, this is hardcoded as http://docker.example.com for SSH.
+	// If we are using a remote host via SSH, we do NOT want to overwrite DOCKER_HOST here.
```

Comment:
How can we cover this change with a test?

Reply:
How about a unit test perhaps after factoring the code a bit? Where's the information flowing when you set os.Setenv("DOCKER_HOST", host), what's the code that reads this back, how can we make that connection more obvious?

Reply:
Poking around the codebase. Why is configureDockerClient is even called in Configure.

I see this comment:

```
	// Configure a Docker daemon client here even though we won't use it. We want use the native
	// provider's logic to determine the host address and then set the DOCKER_HOST environment
	// accordingly so the bridged provider picks it up, too.
```

This is very unfortunate. I see this was introduced in https://github.com/pulumi/pulumi-docker/pull/697/files#diff-0e3b5cd2e6abf4633712dbd2d69307b7fbd75bc71844e34306afdcd9840dab64 with very abbreviated testing also.

I'll ask the team.

Reply:
https://github.com/pulumi/pulumi-docker/issues/710